ush/gfs/task/analysis.py

```diff
@@ -0,0 +1,409 @@
+            if os.path.exists(dest):
+                os.remove(dest)
+            logging.info(f'Copying RESTART {bkg_path} to archive aerosol background')
+            shutil.copy(bkg_path, dest)
+
+        #---- add increments to RESTART files
+        self.add_fms_cube_sphere_increments()
+        #---- move increments to ROTDIR
+        fms_inc_file_template = os.path.join(self.datadir, 'anl', f'aeroinc.{cdate_fv3}.fv_tracer.res.tile1.nc')
+        for itile in range(1,7):
+            inc_path = fms_inc_file_template.replace('tile1', f'tile{itile}')
+            dest = os.path.join(os.environ['COMOUTaero'], os.path.basename(inc_path))
+            if os.path.exists(dest):
+                os.remove(dest)
+            logging.info(f'Copying aerosol FMS cube sphere increment to {dest}')
+            shutil.copy(bkg_path, dest)
+
+    def add_fms_cube_sphere_increments(self):
+        import netCDF4 as nc
+        logging.info('Adding increments to RESTART files')
+        # only need the fv_tracer files
+        cdate_fv3 = dt.datetime.strptime(self.cdate, '%Y%m%d%H').strftime('%Y%m%d.%H%M%S')
+        fms_inc_file_template = os.path.join(self.datadir, 'anl', f'aeroinc.{cdate_fv3}.fv_tracer.res.tile1.nc')
+        comin_ges = os.environ['COMIN_GES']
+        # NOTE that while 'chem' is the $componenet, the aerosol fields are with the 'atmos' tracers
+        comin_ges_atm = comin_ges.replace('chem', 'atmos')
+        fms_bkg_file_template = os.path.join(comin_ges_atm, 'RESTART', f'{cdate_fv3}.fv_tracer.res.tile1.nc')
+        incvars = ['dust1', 'dust2', 'dust3', 'dust4', 'dust5',
+                   'seas1', 'seas2', 'seas3', 'seas4',
+                   'so4', 'oc1', 'oc2', 'bc1', 'bc2']
+        for itile in range(1,7):
+            inc_path = fms_inc_file_template.replace('tile1', f'tile{itile}')
+            bkg_path = fms_bkg_file_template.replace('tile1', f'tile{itile}')
+            with nc.Dataset(inc_path, mode='r') as incfile:
+                logging.info(f'Opening increment file {inc_path}')
+                with nc.Dataset(bkg_path, mode='a') as rstfile:
+                    logging.info(f'Opening RESTART file {bkg_path}')
+                    for vname in incvars:
+                        logging.info(f'...Adding increment for {vname}')
+                        increment = incfile.variables[vname][:]
+                        bkg = rstfile.variables[vname][:]
+                        anl = bkg + increment
+                        rstfile.variables[vname] = anl[:]
+
+    def stage_crtm(self, filedict):
+        logging.info('Staging CRTM coefficient files')
+        self.stage(filedict)
+        logging.info('Finished staging CRTM coefficient files')
+
+    def stage_exe(self):
+        logging.info('Staging FV3-JEDI executable')
+        src = os.environ['JEDIVAREXE']
+        dest = os.path.join(self.datadir, os.path.basename(src))
+        if os.path.exists(dest):
+            os.remove(dest)
+        os.symlink(src, dest)
```

Comment:
We use a link for the executable.   Not sure if NCO will accept this.  I think they want a physical copy of everything in the run directory.  This allows them to tar up a run directory and ship it to GDIT or others for diagnostics, evaluation, or testing.

Reply:
@RussTreadon-NOAA , you can tar the files that the symlink point to with the `tar --dereference` option, so presumably there is an other reason? I'd be curious to know what it is.

Reply:
Eventually we will copy, I've noticed in the past (perhaps this has been fixed) that copying JEDI executables results in issues with library links @guillaumevernieres have you noticed a similar issue? Perhaps that is something from the 2019 era that is no longer an issue.

Reply:
@CoryMartin-NOAA , not that I can remember. I usually create a link or point directly to the executable.

Reply:
I think another reason has to do with the design of WCOSS2.   WCOSS2 has physical disks (HDD) and solid state disks (SSD).  Operations runs jobs on SSD.   The input and output for these jobs reside in HDD.   NCO wants everything copied to the WCOSS2 SSD run directories to ensure optimal performance by having executables only use SSD for runtime i/o.    Did I get this right, @CoryMartin-NOAA

Reply:
That is my understanding yes

Reply:
Thanks, @CoryMartin-NOAA